lua/effects/oneillbeam.lua

```diff
@@ -30,7 +30,7 @@ EFFECT.RefractBeam = Material("models/shadertest/predator");
 
 function EFFECT:Init(data)
 	self.StartPos	= data:GetOrigin();
-	self.TargetEnt	= StarGate.FindEntInsideSphere(self.StartPos, 10, "ring_base_ancient")[1];	-- Remote Gate
```

Comment:
Are you sure this is right? Isn't laser beam available only for goauld rings?

Reply:
The laser easter egg is [defined in the base ring](https://github.com/RafaelDeJongh/cap/blob/master/lua/entities/ring_base/init.lua#L527-L546), so it *should* be available for entities using it as a base. It's just that only the ancient ring [seems to use it](https://github.com/RafaelDeJongh/cap/blob/master/lua/entities/ring_panel.lua#L144-L145).

Reply:
Tried it, working on goauld too

Reply:
This is likely because the [origin of the ring laser is positioned in the centre of the ring's model](https://github.com/RafaelDeJongh/cap/blob/master/lua/entities/ring_base/init.lua#L534), and since the Ori rings have a platform while the others don't, the ring is inside the model and thus glitching out/not being visible at all.

A hacky fix could just be to check if the entity is `ring_base_ori` and if it is, give it a slight offset on the +Z axis (this would also need to be world-to-local'd to prevent positioning errors when the ring is rotated).

Reply:
Yeah, but don't think it's really useful, Ori base his bigger than other and moving it while laser is fired could possibily cause few problems